tenpy/algorithms/truncation.py

```diff
@@ -534,6 +534,8 @@ def decompose_theta_qr_based(old_T_L: npc.Array, old_T_R: npc.Array, theta: npc.
 
     if compute_err:
         return_both_T = True
+    
+    theta /= npc.norm(theta) # <- TODO: discuss this, since for TEBD already normalized. Alternatively, normalize in VariationalCompression.update_local()
```

Comment:
We need to normalize theta somewhere if we are interested in the error. 

Does it make sense to do this in general right at the beginning, only when we need it [before calculating the error](https://github.com/tenpy/tenpy/blob/ce3ac8da60d4bf02b6a1fae5855cfd072f1c2cf9/tenpy/algorithms/truncation.py#L600) or maybe better directly in [VariationalApplyMPO.update_local()](https://github.com/tenpy/tenpy/blob/ce3ac8da60d4bf02b6a1fae5855cfd072f1c2cf9/tenpy/algorithms/mps_common.py#L2382)?

Reply:
We take care of normalization when we decompose Xi.
There is no need to normalize theta beforehand.
Should document this.
Maybe the pictures should say something like ``theta == renormalize * T_Lc---T_Rc``.
Also, should document the ``S``  and ``renormalize`` return values accordingly.

Reply:
Yes, that is true. However, the error is currently computed as `eps = npc.norm(theta-theta_approx)**2` and in general `npc.norm(theta)!=1`, but `npc.norm(theta_approx)=1`. We do get the `renormalization` from `svd_theta` or `_eig_based_svd()`, but this is not the norm of `theta`; in fact, it is a combination of `npc.norm(theta)` and `new_norm` from `truncate(S, trunc_par)`. This means to calculate the error `eps`, we still need to compute `npc.norm(theta)` somewhere. After today's meeting we concluded that it is fine to just compute the norm directly where it is needed, e.g. `eps = npc.norm(theta/npc.norm(theta)-theta_approx)**2` .